server/src/main/java/org/opensearch/index/store/lockmanager/FileLockInfo.java

```diff
@@ -88,21 +87,18 @@ static String generateLockName(String fileToLock, String acquirerId) {
         }
 
         public static String getFileToLockNameFromLock(String lockName) {
-            String[] lockNameTokens = lockName.split(RemoteStoreLockManagerUtils.SEPARATOR);
-
-            if (lockNameTokens.length != 2) {
+            if (!lockName.endsWith(".lock")) {
                 throw new IllegalArgumentException("Provided Lock Name " + lockName + " is not Valid.");
             }
-            return lockNameTokens[0];
+            return lockName.replace(getAcquirerIdFromLock(lockName) + RemoteStoreLockManagerUtils.LOCK_FILE_EXTENSION, "");
         }
 
         public static String getAcquirerIdFromLock(String lockName) {
-            String[] lockNameTokens = lockName.split(RemoteStoreLockManagerUtils.SEPARATOR);
-
-            if (lockNameTokens.length != 2) {
+            if (!lockName.endsWith(".lock")) {
                 throw new IllegalArgumentException("Provided Lock Name " + lockName + " is not Valid.");
             }
-            return lockNameTokens[1].replace(RemoteStoreLockManagerUtils.LOCK_FILE_EXTENSION, "");
+            String[] lockNameTokens = lockName.split(RemoteStoreLockManagerUtils.SEPARATOR);
+            return lockNameTokens[lockNameTokens.length - 1].replace(RemoteStoreLockManagerUtils.LOCK_FILE_EXTENSION, "");
```

Comment:
We will have similar issue in RemoteStore metadata file name. @sachinpkale Fyi

Reply:
for new versions, using '...' as the separator since '.' is not there in url safe base64 character set. keeping support for old version locks as well but if its old lock file, fetching the last index of `.lock` substring and returning the substring between these index.

Reply:
For segment and translog metadata file names, we don't have UUIDs there , so this shouldn't be an issue there.